File: project/5.04_コンストラクタのオーバーロード/java/java_ex5_04_1/src/java_ex5_04_1/Ex5_04_1.java
package java_ex5_04_1;

class Circle {
	double cx;			// 中心座標x
	double cy;			// 中心座標y
	double radius;		// 半径
	
	Circle ( double arg_radius ){
		cx = 0.0;
		cy = 0.0;
		radius = arg_radius;
	}
	
	Circle ( double arg_cx, double arg_cy, double arg_radius ){
		cx = arg_cx;
		cy = arg_cy;
		radius = arg_radius;
	}
	
	Circle ( String arg_radius ){
		cx = 9.9;
		cy = 8.8;
		radius = Double.valueOf( arg_radius );
	}
	
}

public class Ex5_04_1 {

	public static void main(String[] args) {
		// シグネチャがdouble引数１つのコンストラクタが実行される
		Circle c1 = new Circle( 5.5 );
		System.out.println( c1.cx + ", " + c1.cy + ", " + c1.radius );
		
		// シグネチャがdouble引数３つのコンストラクタが実行される
		Circle c2 = new Circle( 1.2, 3.4, 5.5 );
		System.out.println( c2.cx + ", " + c2.cy + ", " + c2.radius );
		
		// シグネチャがString引数１つのコンストラクタが実行される
		Circle c3 = new Circle( "5.5" );
		System.out.println( c3.cx + ", " + c3.cy + ", " + c3.radius );
	}

}

File: project/5.04_コンストラクタのオーバーロード/py/py_ex5_04_1/src/ex5_04_1.py
'''
Created on 2023/05/01

@author: Keroichi.T
'''

# pythonにオーバーロードはない
class Circle:
    # デフォルト値のない引数を、ある引数よりも後に記述できない
    # def __init__(self, arg_cx = 0.0, arg_cy = 0.0, arg_radius):

    def __init__(self, arg_cx = 0.0, arg_cy = 0.0, arg_radius = None):
        self.cx = arg_cx
        self.cy = arg_cy
        self.radius = arg_radius

    # def __init__(self, arg_cx):
    #     self.cx = arg_cx


def main():
    # 引数を指定して設定することができる
    c1 = Circle( arg_radius = 5.5 )
    print( "{}, {}, {}".format(c1.cx, c1.cy, c1.radius ) )

    c2 = Circle( 1.2, 3.4, 5.5 )
    print( "{}, {}, {}".format(c2.cx, c2.cy, c2.radius ) )
    
    c3 = Circle( 1.2, 3.4 )
    print( "{}, {}, {}".format(c3.cx, c3.cy, c3.radius ) )
    

if __name__ == '__main__':
    main()